crud/select/filters.lua

```diff
@@ -40,11 +40,15 @@ local function is_early_exit_possible(index, tarantool_iter, condition)
     return false
 end
 
-local function get_index_fieldnos(index)
+local function get_index_fieldnos(index, space_format)
     local index_fieldnos = {}
 
     for _, part in ipairs(index.parts) do
-        table.insert(index_fieldnos, part.fieldno)
+        if part.path ~= nil then
+            table.insert(index_fieldnos, string.format("%s.%s", space_format[part.fieldno].name, part.path))
```

Comment:
Hmmm... Could you please check some complex "path". I assume that path could contain spaces and special characters... Probably correct variant is `string.format(`[%q][%q]`, space_format[part.fieldno].name, part.path))`

Reply:
Also I think you don't need `space_format` here.  `part` contains field (or fieldno) and it should be enough.

Reply:
If we get ``part.path`` = car.model and ``fieldno`` = 5 (part doesn't contain``field``), we'll get ["5"]["car.model"]. We can split ``part.path`` by point, but I don't that's a gooid idea...

Reply:
Maybe such way could consider each possible combination...

```lua
tarantool> t = box.tuple.new({1, 2, {car = {model = 'bmw'}}})
---
...

tarantool> t['[3]car.model']
---
- bmw
...

tarantool> t['[3].car.model']
---
- bmw
...

tarantool> t['[3]["car"]["model"]']
---
- bmw
...
```

Anyway suggested solution seems doesn't solve a problem with such paths:
```lua
tarantool> box.space.space:create_index('json2', {parts = {{field = 2, path = '["hello world"]', type = 'string'}}})
---
- unique: true
  parts:
  - type: string
    is_nullable: false
    path: '["hello world"]'
    fieldno: 2

tarantool> box.space.space:create_index('json', {parts = {{field = 2, path = '["car"]["model"]', type = 'string'}}})
---
- unique: true
  parts:
  - type: string
    is_nullable: false
    path: '["car"]["model"]'
    fieldno: 2
```